xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-rss/src/main/java/org/xwiki/rendering/internal/macro/rss/RssMacro.java

```diff
@@ -112,29 +121,83 @@ public boolean supportsInlineMode()
         return false;
     }
 
+    /**
+     * Allow to save the current processed feed in the context, to be able to retrieve without concurrency issue.
+     * @param feed the feed to save in the current context.
+     */
+    private void setFeed(SyndFeed feed)
+    {
+        this.execution.getContext().setProperty(FEED_PROPERTY, feed);
+    }
+
+    /**
+     * Retrieve the feed of the current context.
+     * @return the feed that is being processed in the macro.
+     */
+    private SyndFeed getFeed()
+    {
+        return (SyndFeed) this.execution.getContext().getProperty(FEED_PROPERTY);
+    }
+
+    /**
+     * Remove the feed information from the current context.
+     */
+    private void removeContextFeed()
+    {
+        this.execution.getContext().removeProperty(FEED_PROPERTY);
+    }
+
+    @Override
+    protected ResourceReference getImageReference(RssMacroParameters parameters, String content,
+        MacroTransformationContext context)
+    {
+        if (parameters.isDecoration() && parameters.isImage()) {
+            return new ResourceReference(getFeed().getImage().getUrl(), ResourceType.URL);
```

Comment:
The macro used to check `feed.getImage() != null` but this seems to be vanished.